### PR TITLE
feat: use trpc 11

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #<issue>

## ✅ Checklist

- [ ] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [ ] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I performed a functional test on my final commit

---

## Changelog

_[Short description of what has changed]_

---

## Screenshots

_[Screenshots]_

💯

### CODE REVIEW

**Review by c-ehrlich — COMMENTED** [review]
pr looks good. how do you feel about merging while trpc v11 is still in beta? from what i understand APIs/interfaces aren't guaranteed to be stable yet?

**Comment by crivera** [comment]
Is there a way to use this without being merged? `npm create t3-app@trpc11`?

Thank you

File: cli/template/extras/src/trpc/react.tsx
Comment (DanielSintimbrean) [inline cli/template/extras/src/trpc/react.tsx:27]:
```suggestion
          transformer: SuperJSON,
          async headers() {
```

Apply [Transformers are moved to links (breaking)](https://trpc.io/docs/migrate-from-v10-to-v11#transformers-are-moved-to-links-breaking) migration

**Comment by CuriouslyCory** [comment]
https://github.com/t3-oss/create-t3-app/blob/ec56d53245f2b1579d088568db5154211ffd056f/cli/template/extras/src/utils/api.ts#L29
`transformer: superjson` needs to be moved to the httpLinkBatch and outside of the config() function. 
Here's an example of a working config.
```ts
/**
 * This is the client-side entrypoint for your tRPC API. It is used to create the `api` object which
 * contains the Next.js App-wrapper, as well as your type-safe React Query hooks.
 *
 * We also create a few inference helpers for input and output types.
 */
import { httpBatchLink, loggerLink } from "@trpc/client";
import { createTRPCNext } from "@trpc/next";
import { type inferRouterInputs, type inferRouterOutputs } from "@trpc/server";
import superjson from "superjson";

import { type AppRouter } from "~/server/api/root";

const getBaseUrl = () => {
  if (typeof window !== "undefined") return ""; // browser should use relative url
  if (process.env.VERCEL_URL) return `https://${process.env.VERCEL_URL}`; // SSR should use vercel url
  return `http://localhost:${process.env.PORT ?? 3000}`; // dev SSR should use localhost
};

/** A set of type-safe react-query hooks for your tRPC API. */
export const api = createTRPCNext<AppRouter>({
  config() {
    return {
      /**
       * Links used to determine request flow from client to server.
       *
       * @see https://trpc.io/docs/links
       */
      links: [
        loggerLink({
          enabled: (opts) =>
            process.env.NODE_ENV === "development" ||
            (opts.direction === "down" && opts.result instanceof Error),
        }),
        httpBatchLink({
          url: `${getBaseUrl()}/api/trpc`,
          transformer: superjson,
        }),
      ],
    };
  },
  /**
   * Transformer used for data de-serialization from the server.
   *
   * @see https://trpc.io/docs/data-transformers
   */
  transformer: superjson,
  /**
   * Whether tRPC should await queries when server rendering pages.
   *
   * @see https://trpc.io/docs/nextjs#ssr-boolean-default-false
   */
  ssr: false,
});

/**
 * Inference helper for inputs.
 *
 * @example type HelloInput = RouterInputs['example']['hello']
 */
export type RouterInputs = inferRouterInputs<AppRouter>;

/**
 * Inference helper for outputs.
 *
 * @example type HelloOutput = RouterOutputs['example']['hello']
 */
export type RouterOutputs = inferRouterOutputs<AppRouter>;
```

**Comment by alexander-densley** [comment]
I used this new create t3 app and am trying to integrate clerk auth, unfortunately it is not working, despite it working on the previous t3 set up. here is the link to the gist where clerk was working: https://gist.github.com/mkcode/a590d1c8f7b0a37b8299965de7f7e958

Any advice on how to implement this properly with the new t3 set up?

**Comment by juliusmarminge** [comment]
Check this out for a clerk example https://github.com/juliusmarminge/t3-appdir-clerk

**Comment by markomitranic** [comment]
@juliusmarminge is there a goal for when this will get merged? 

Asking simply because not sure if we will be waiting for trpc to exit beta before its merged - in which case I won't wait for it and will adopt this PR manually in my project :)

**Comment by nvwebd** [comment]
Is there a plan or maybe an ETA for this? I've basically copied the changes into my codebase :D

Maybe a waaay of crazy question but maybe a dedicated branch for all the crazy beta unstable stuff? Since Auth.js is also in experimental mode ( and I already went on latest with it ).

Cheers and regards from Austria.

**Comment by stunaz** [comment]
Not sure why prisma is mentioned here.I am using the latest of everything… auth 5, trpc 11, react-query 5… all is good , been using this PR

**Comment by juliusmarminge** [comment]
This'll ship as soon as someone gets the opportunity to review it and approve it

**Review by c-ehrlich — APPROVED** [review]
this looks great, so much cleaner than the v10 equivalent. if you believe v11 beta is good enough to ship, i'm happy to merge :)

**Comment by alexander-densley** [comment]
question, with this, is there a best practice to enable ssr for prefetching data? I noticed that here we use createTRPCReact instead of createTRPCNext  so we cant just enable ssr. Thanks for all the hard work, yall are the best!

**Comment by maiconcarraro** [comment]
> question, with this, is there a best practice to enable ssr for prefetching data? I noticed that here we use createTRPCReact instead of createTRPCNext so we cant just enable ssr. Thanks for all the hard work, yall are the best!

@alexander-densley I could be wrong, but pretty sure `createTRPCReact` is to enable tanstack/react-query for client side, for the server it's using the `createCallerFactory`

**Comment by juliusmarminge** [comment]
No the ”bridge” between RQ and RSC data hasnt been closed in this update. We’re still experimenting with how this could look (for example, see https://github.com/trpc/trpc/pull/5451)

**Pull request merged**
